cmd/storage-rest-server.go

```diff
@@ -285,6 +281,34 @@ func (s *storageRESTServer) ReadFileHandler(w http.ResponseWriter, r *http.Reque
 	w.Write(buf)
 }
 
+// ReadFileHandler - read section of a file.
+func (s *storageRESTServer) ReadFileStreamHandler(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		return
+	}
+	vars := mux.Vars(r)
+	volume := vars[storageRESTVolume]
+	filePath := vars[storageRESTFilePath]
+	offset, err := strconv.Atoi(vars[storageRESTOffset])
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+	length, err := strconv.Atoi(vars[storageRESTLength])
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+	rc, err := s.storage.ReadFileStream(volume, filePath, int64(offset), int64(length))
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+	defer rc.Close()
+	w.Header().Set("Content-Length", strconv.Itoa(length))
+	io.Copy(w, rc)
```

Comment:
I don't see `r.ContentLength` on the rest/client being used you can simply do `io.LimitReader(rc, length)` ?

Reply:
If the http client raises error if server sends data that is not equal to r.ContentLength then it will be useful, otherwise not. (For example I have seen aws-sdk saying "server returned less data than what it advertised in Content-Length") But I am not sure if golang-http-client does it. Even otherwise it's good to set it since http servers set it when the content-length is known, (it can be helpful in HTTP_TRACE)